src/Cli/dotnet/SlnProjectExtensions.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Sln.Internal;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    internal static class SlnProjectExtensions
+    {
+        public static string GetFullSolutionFolderPath(this SlnProject slnProject)
+        {
+            var slnFile = slnProject.ParentFile;
+            var nestedProjects = slnFile.Sections
+                .GetOrCreateSection("NestedProjects", SlnSectionType.PreProcess)
+                .Properties;
```

Comment:
Is there any chance the project will be in `.NestedPropertySets` vs `.Properties`? I couldn't figure out what the NestedPropertySets were for. If you aren't sure, please feel free to ping me or those on the SLN team so they can provide an answer here.

Reply:
From talking to Stanton and reviewing the code, I think that the NestedPropertySets are only used for the `ProjectConfigurationPlatforms` section. You can have the configuration and the ActiveCfg property nested in a single line. https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/Microsoft.DotNet.Cli.Sln.Internal/SlnFile.cs#L88

You can see where we load property sets we expect to have them in a form that includes a . which fits with the above assumption: https://github.com/dotnet/sdk/blob/a30e465a2e2ea4e2550f319a2dc088daaafe5649/src/Cli/Microsoft.DotNet.Cli.Sln.Internal/SlnFile.cs#L581

So in the case of the NestedProjects section, using Properties and ignoring NestedPropertySets makes sense.